### ISSUE TITLE
[FEAT] Add wandb logging to training code 

### ISSUE THREAD

**Issue by jihyun-0611** [boostcampaitech7/level2-cv-datacentric-cv-23#5]
### 📄 만들고자 하는 기능

- train.py 에 wandb log 코드 추가 
- 이후 wandb UI에서 학습 추적 

### 🌱 구현 내용

- [x] wandb init 에 실험 태그 및 노트 추가 
- [x] wandb config에 추적하고 싶은 하이퍼파라미터 (learning rate, epoch, image size, batch size) 추가 
- [x] wandb log를 통해 학습 메트릭 추적 
(epoch 기준 메트릭 : epoch_angle_loss, epoch_cls_loss, epoch_iou_loss, epoch_loss, elapsed_time) 
(batch 기준 메트릭 :  batch_loss)
- [ ] 추후 쉘 스크립트를 통한 학습 실행을 위해 wandb 관련 argument 추가 

### 📅 예상 기간

2024-10-30 ~ 2024-10-30

### 📝 기타

- !pip install wandb 
- wandb login 
--> api-key : [wandb.ai/authorize](https://wandb.ai/authorize )
→ 위 링크에서 본인 계정의 api key 복사
→ 처음 한 번만 해당 api key 입력하여 로그인
